data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/datatype/MySQLDataType.java

```diff
@@ -0,0 +1,176 @@
+package org.opensearch.dataprepper.plugins.source.rds.datatype;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public enum MySQLDataType {
+    // Numeric types
+    TINYINT("tinyint", DataCategory.NUMERIC, DataSubCategory.SIGNED),
+    TINYINT_UNSIGNED("tinyint unsigned", DataCategory.NUMERIC, DataSubCategory.UNSIGNED),
+    SMALLINT("smallint", DataCategory.NUMERIC, DataSubCategory.SIGNED),
+    SMALLINT_UNSIGNED("smallint unsigned", DataCategory.NUMERIC, DataSubCategory.UNSIGNED),
+    MEDIUMINT("mediumint", DataCategory.NUMERIC, DataSubCategory.SIGNED),
+    MEDIUMINT_UNSIGNED("mediumint unsigned", DataCategory.NUMERIC, DataSubCategory.UNSIGNED),
+    INT("int", DataCategory.NUMERIC, DataSubCategory.SIGNED),
+    INT_UNSIGNED("int unsigned", DataCategory.NUMERIC, DataSubCategory.UNSIGNED),
+    BIGINT("bigint", DataCategory.NUMERIC, DataSubCategory.SIGNED),
+    BIGINT_UNSIGNED("bigint unsigned", DataCategory.NUMERIC, DataSubCategory.UNSIGNED),
+    DECIMAL("decimal", DataCategory.NUMERIC, DataSubCategory.SIGNED),
+    FLOAT("float", DataCategory.NUMERIC, DataSubCategory.SIGNED),
+    DOUBLE("double", DataCategory.NUMERIC, DataSubCategory.SIGNED),
+    BIT("bit", DataCategory.NUMERIC, DataSubCategory.BIT),
+
+    // String types
+    CHAR("char", DataCategory.STRING, DataSubCategory.CHAR),
+    VARCHAR("varchar", DataCategory.STRING, DataSubCategory.CHAR),
+    TINYTEXT("tinytext", DataCategory.STRING, DataSubCategory.BYTES),
```

Comment:
Wondering if what's the difference between CHAR and BYTES sub category. Are they represented differently in binlog events?

Reply:
The once tagged with `DataSubCategory.BYTES` the value is encoded in bytes in binlogs.

Reply:
Is VARCHAR encoded in bytes in binlog as well, or not?

Reply:
No, CHAR and VARCHAR are not encoded in bytes